### PR TITLE
Titans features

### PR DESCRIPTION
**Pull request by titanlab**
Contains all changes I made. Also contains sync with recent changes from official branch.

### CODE REVIEW

**Comment by MBeijer** [comment]
I'll review this after work :)

File: src/tracker/SectionSettings.cpp [hunk 1]
```diff
@@ -176,6 +176,7 @@ enum ControlIDs
 	BUTTON_RESOLUTIONS_CUSTOM,
 	BUTTON_RESOLUTIONS_FULL,
 	STATICTEXT_SETTINGS_MAGNIFY,
+    STATICTEXT_SETTINGS_WINDOWED_ONLY,
```
Comment (MBeijer) [inline src/tracker/SectionSettings.cpp:179]:
```suggestion
	STATICTEXT_SETTINGS_WINDOWED_ONLY,
```

File: src/tracker/amigaversion.h.in [hunk 2]
```diff
@@ -1,3 +1,4 @@
 static const char __attribute((used)) *amiga_ver = "$VER: MilkyTracker ${VER_X}.${VER_YY} (${VER_DAY}.${VER_MONTH}.${VER_YEAR}) MilkyTracker v${VER_X}.${VER_YY}.${VER_ZZ} ported by AmigaDev.com Team\0";
```
Comment (MBeijer) [inline src/tracker/amigaversion.h.in:1]: this is VER_YEAR etc is used from main CMakeLists.txt

**Review by MBeijer — CHANGES_REQUESTED** [review]
I've now gone through everything and asked some questions and added some suggestions

File: CMakeLists.txt [hunk 3]
```diff
@@ -47,25 +51,12 @@ endif()
...
 # Version number in format X.YY.ZZ
 set(VER_X  1)
 set(VER_YY 03)
 set(VER_ZZ 00)
 set(VER_FULL "${VER_X}.${VER_YY}.${VER_ZZ}")
 
-# Build date Information
-string(TIMESTAMP VER_YEAR "%Y")
-string(TIMESTAMP VER_MONTH "%m")
-string(TIMESTAMP VER_DAY "%d")
```
[the remark this answers is not in the document]
Comment (titanlab) [reply CMakeLists.txt:62]: Right, will add that again.

File: cmake/CPackLHA.cmake [hunk 4]
```diff
@@ -0,0 +1,5 @@
+find_program(LHA_EXECUTABLE lha)
```
[the remark this answers is not in the document]
Comment (titanlab) [reply cmake/CPackLHA.cmake:1]: No, lol

File: .gitmodules [hunk 5]
```diff
@@ -2,12 +2,12 @@
...
+	ignore = dirty
 [submodule "lhasa"]
 	path = src/submodules/lhasa
 	url = https://github.com/fragglet/lhasa.git
+	ignore = dirty
 [submodule "zlib"]
 	path = src/submodules/zlib
 	url = https://github.com/madler/zlib.git
-[submodule "zziplib"]
-	path = src/submodules/zziplib
```
[the remark this answers is not in the document]
Comment (titanlab) [reply .gitmodules:12]: Will add it again.

File: include/amiga/clib/cybergraphics_protos.h [hunk 6]
```diff
@@ -0,0 +1,53 @@
+/*
```
[the remark this answers is not in the document]
Comment (titanlab) [reply include/amiga/clib/cybergraphics_protos.h:1]: Yep, right. :-)

File: src/tracker/amigaversion.h.in (hunk 2 above)
Comment (titanlab) [reply src/tracker/amigaversion.h.in:1]: OK

File: src/ppui/sdl/DisplayDeviceFB_SDL.cpp [hunk 7]
```diff
@@ -31,26 +31,48 @@
 #include "DisplayDeviceFB_SDL.h"
 #include "Graphics.h"
 
+#include <unistd.h>
+
+#if defined(AMIGA_SAGA_PIP)
```
[the remark this answers is not in the document]
Comment (titanlab) [reply src/ppui/sdl/DisplayDeviceFB_SDL.cpp:36]: Will remove the SAGA implemention from the SDL framebuffer device. I had that in the first prototype.

**Review by MBeijer — APPROVED** [review]

**Review by MBeijer — APPROVED** [review]

**Review by MBeijer — APPROVED** [review]

**Pull request merged**